app/controllers/laboratory_samples_controller.rb

```diff
@@ -27,14 +27,13 @@ def create
 
     @sample = LaboratorySample.new(sample_params.merge(uuid: uuid))
 
-    if @sample.save
-      session.delete(:creating_sample_uuid)
+    @sample.save!
+    session.delete(:creating_sample_uuid)
+    redirect_to laboratory_samples_path, notice: 'Sample was successfully created.'
 
-      next_url = laboratory_samples_path(@sample)
-      redirect_to next_url, notice: 'Sample was successfully created.'
-    else
-      render action: 'new'
-    end
+  rescue ActiveRecord::RecordInvalid => ex
+    @sample.errors.add :sample_type, ex.message
+    render action: 'new'
```

Comment:
[Using exceptions for normal control flow is usually not a good idea](https://wiki.c2.com/?DontUseExceptionsForFlowControl).

`ActiveRecord::Model#save` already populates the `.errors` array. So I understand this change is not necessary at all.

This code should do:

```
@sample = LaboratorySample.new(sample_params.merge(uuid: uuid))

    if @sample.save
      session.delete(:creating_sample_uuid)
      redirect_to laboratory_samples_path, notice: 'Sample was successfully created.'
    else
      render action: 'new'
    end
```

Why would you change that into a `save!`?

Reply:
I'd also redirect users to `edit_laboratory_sample_path(@sample)` on success, so they can see & print the barcode.